### ISSUE TITLE
Error when creating docker container: no such host

### ISSUE THREAD

**Issue by imulab** [ory/dockertest#78]
(Docker version is `1.13.1`)

I was using the example MongoDB code to create a mongo container in my test case and encountered this no such host error when connecting:

```
Post http://unix.sock/containers/create?: dial tcp: lookup unix.sock: no such host
```

Suspected a OS X platform issue, I also tried to run the test case on Travis CI with the following configuration:
```
language: go
sudo: required

go:
- 1.7

services:
- docker
```
but got the same error.

Here is the code that caused the problem, it is essentially same with the example code:
```go
func connectToDockerMongo(t *testing.T) (*dockertest.Pool, *dockertest.Resource) {
	pool, err := dockertest.NewPool("")
	fatalIfError(t, err)

	dockerResource, err := pool.Run("mongo", "3.3", []string{""})
	fatalIfError(t, err)

	err = pool.Retry(func() error {
		var e error
		session, e := mgo.Dial(fmt.Sprintf("localhost:%s", dockerResource.GetPort("27017")))
		if e != nil {
			t.Error(e)
			return e
		}
		return session.Ping()
	})
	fatalIfError(t, err)

	return pool, dockerResource
}
```
Basically, `pool.Run("mongo", "3.3", []string{""})` is failing.

Could someone please help. Thanks!

**Comment by aeneasr** [comment]
Looks like the socket changed (not sure to what, I would have to investigate). Try connecting with `NewPool("http://localhost:2375")`

**Comment by imulab** [comment]
Connecting to `localhost:2375` yields a fast `cannot connect to Docker endpoint` error.

**Comment by aeneasr** [comment]
Ok, you need to see for yourself where docker is listening on and provide that value as an argument to newpool

**Comment by imulab** [comment]
thx, I ll try that

**Comment by galaxydi** [comment]
How to check the listening file in MAC, I found it's hard to find out where is the docker daemon config on MAC?  

The problem was resovled?

**Comment by calvn** [comment]
Are you using docker-machine, or docker for MacOS? If it's the former, you need to find the IP of the VM and connect to it instead of localhost. Hope this helps!

**Comment by pbarker** [comment]
Just hit this as well, if you try an test hydra in an alpine container this breaks it.

**Comment by aeneasr** [comment]
I'm hitting the same issue on Windows now, will investigate soon.

**Comment by pbarker** [comment]
So mine ended up being that I needed `network_mode: "host"`, but that is currently not working on a mac seen here https://github.com/docker/for-mac/issues/1031. I just found a workaround for the time being.

**Comment by aeneasr** [comment]
Well, you simply need to expose http in the docker general config:

![image](https://cloud.githubusercontent.com/assets/3372410/26826297/dc5344fe-4ab8-11e7-8a78-330c5b8025aa.png)

We should probably investigate adding TLS soon here.

**Comment by pbarker** [comment]
must be a windows thing, don't have that on docker-for-mac yet ¯\_(ツ)_/¯

**Comment by pinkisemils** [comment]
OSX won't require TLS because Docker on OSX does not use a TCP socket but rather a Unix socket that is only accessible by local processes anyway and all of the unixy file permissions apply.

**Comment by aeneasr** [comment]
right, that makes sense! so does it work now for everyone?

**Comment by au-phiware** [comment]
Why not use [`NewClientFromEnv`](https://godoc.org/github.com/fsouza/go-dockerclient#NewClientFromEnv)?

**Comment by rugwirobaker** [comment]
I am having this same issue while running tests in circleci. If I am not mistaken it clearly can connect to the docker daemon but can not start the container. since if fails on this line:

`container, err := pool.Run("postgres", "alpine", config)`
`	if err != nil {`
`		log.Fatalf("Could not start container: %s", err)`
}`
`
**returning:**

`Could not start container: : dial unix /var/run/docker.sock: connect: no such file or directory`

**environment**: `circleci linux`

**Comment by aeneasr** [comment]
Enable Docker in Circle.

**Comment by rugwirobaker** [comment]
I have been working on it and the problems seems to be that circleci allows you yo run docker commands only when you setup a local docker host. In my tests am connecting to localhost but the remote docker daemon is of course remote. I have tried using the `DOCKER_HOST` to no effect. S does anyone know if there is a way you can proxy your local clients requests to a remote docker without having to use it's address directly??

**Comment by fanadol** [comment]
I also experience this on gitlab ci.

**Comment by avarghese-sqsp** [comment]
Is this still a problem?

**Comment by AlexeyInc** [comment]
Still have above problem 

Getting ` dial unix /var/run/docker.sock: connect: no such file or directory` when trying to run docker container with dockertest on mac M1

**Comment by farshadahmadi** [comment]
> Still have above problem
> 
> Getting ` dial unix /var/run/docker.sock: connect: no such file or directory` when trying to run docker container with dockertest on mac M1

Got the same issue today! Any solution?

**Comment by robmonte** [comment]
I was seeing the same error as well. For me, going into the settings and then the advanced settings section of Docker Desktop, I enabled the option `Allow the default Docker socket to be used (requires password)`. Afterwards my tests worked again.
